### PR TITLE
DELAY_BETWEEN_EACH_EMAIL

### PR DESCRIPTION
**Pull request by mullerivan**
:8ball:

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -69,6 +69,12 @@ Installation
 
     If not set, django-newsletter will fall back to Django's default TextField
     widget.
+#)  Configure delay between each Email (optional).
+
+    As default the configuration delay between each Email is 0 but you can easily
+    change it to what ever you feel is good for your company::
+
+        DELAY_BETWEEN_EACH_EMAIL = 0
```
Comment (dokterbob) [inline docs/installation.rst]: Please make sure here the number is seconds and can be factional (for 10000 emails waiting 0.1 or 1 second makes a huge difference. Also, if you can, please write Email as email.

File: newsletter/models.py [hunk 2]
```diff
@@ -569,6 +569,7 @@ def extra_headers(self):
         }
 
     def submit(self):
+        import time
```
Comment (dokterbob) [inline newsletter/models.py]: Please keep imports at the top of the file, if you can.

File: newsletter/models.py [hunk 3]
```diff
@@ -584,6 +585,8 @@ def submit(self):
 
         try:
             for subscription in subscriptions:
+                if hasattr(settings, 'DELAY_BETWEEN_EACH_EMAIL'):
+                    time.sleep(settings.DELAY_BETWEEN_EACH_EMAIL)
```
Comment (dokterbob) [inline newsletter/models.py]: Nice and clean. Much more compact. You're nearly there.

**Comment by dokterbob** [comment]
Ok, reviewed it. It looks fine now, except that it would be nice to have a little test for the added functionality, so that it actually waits a specified amount of time (like 0.001 s). (Later, ideally, we would mock patch `time.sleep()` so to remove the delay.)

To update the pull request you don't need to create a new one. Just create a branch for this pull request (name it whatever makes sense to you) and make the requested modifications there. Change the pull request to match the branch. Every time the branch is updated, the pull request is updated too.

To create a branch: `git checkout -b funnybranchname`
To push it `git push -u origin funnybranchname`
(`-u` makes sure that afterwards you can just do `git push` without specifiying the branch)

If you want to get really fancy you can start rebasing, see https://stackoverflow.com/questions/9790448/how-to-update-a-pull-request for an example of how to do that.

**Comment by mullerivan** [comment]
I'm not 100% sure  how to  test the delay between emails......

**Comment by dokterbob** [comment]
Code looks great now, thanks @mullerivan!

To test this feature, you could write a variant of the submission test where you measure the time it takes to submit to two recipients first without an `EMAIL_DELAY` and then with `EMAIL_DELAY` set to a small but measurable value (i.e. 0.1 s).

Then you assert that the difference in execution time is [roughly equal](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertAlmostEqual) to the configured delay.

**Comment by dokterbob** [comment]
This has been implemented and merged in #223.

**Pull request closed without merging**